### ISSUE TITLE
Standalone S3 Lifecycle Rule for ExpiredObjectDeleteMarker=true does not work

### ISSUE THREAD

**Issue by noslowerdna** [aws-cloudformation/cloudformation-coverage-roadmap#1124]
### Name of the resource

AWS::S3::Bucket

### Issue Description

Using the AWS console or CLI we can create a minimal S3 bucket lifecycle rule that sets `ExpiredObjectDeleteMarker` to `true`, with no other functionality included.

**Console**

<img width="655" alt="Screen Shot 2022-03-23 at 9 54 56 AM" src="https://user-images.githubusercontent.com/930946/160154120-6c2359bb-df7c-49b4-9d9e-52b384fa2997.png">

**CLI**

```
aws s3api put-bucket-lifecycle-configuration --bucket some-test-bucket \
--lifecycle-configuration '{"Rules":[{"Expiration":{"ExpiredObjectDeleteMarker":true},"ID":"Cleanup Delete Markers","Filter":{},"Status":"Enabled"}]}'
```

Both methods result in the following rule creation.

**Console**

<img width="1181" alt="Screen Shot 2022-03-25 at 11 03 07 AM" src="https://user-images.githubusercontent.com/930946/160157730-83470bb0-fa51-47b5-88e8-6080ab1d2a55.png">

**CLI**

```
$ aws s3api get-bucket-lifecycle-configuration --bucket some-test-bucket
{
    "Rules": [
        {
            "Expiration": {
                "ExpiredObjectDeleteMarker": true
            },
            "ID": "Cleanup Delete Markers",
            "Filter": {},
            "Status": "Enabled"
        }
    ]
}
```

However based on our recent experience and confirmed by AWS Support, the same standalone `ExpiredObjectDeleteMarker=true` lifecycle rule cannot be created using CloudFormation. Feature #132 (status: Shipped) incorrectly indicates that it can. 

### Expected Behavior

The CloudFormation stack creation or update succeeds and applies a minimal `ExpiredObjectDeleteMarker=true` lifecycle rule to the S3 bucket.

### Observed Behavior

The CloudFormation stack creation or update fails with this error,

> "At least one of [ExpirationDate,ExpirationInDays,AbortIncompleteMultipartUpload,Transition,Transitions,NoncurrentVersionExpirationInDays,NoncurrentVersionTransition,NoncurrentVersionTransitions,NoncurrentVersionExpiration] needs to be specified"

### Test Cases

```
Resources:
  S3Bucket:
    Properties:
      LifecycleConfiguration:
        Rules:
        - Id: Cleanup Delete Markers
          ExpiredObjectDeleteMarker: true
          Status: Enabled
```

**Comment by noslowerdna** [comment]
Can this bug be prioritized? Thanks!

**Comment by urda** [comment]
> Can this bug be prioritized? Thanks!

I agree. As a former Amazonian It really disappoints me that AWS is not treating this as customer-impact, is not treating this as a `SEV2`.

**Comment by Ragul29** [comment]
Can this bug be prioritized ?

**Comment by urda** [comment]
@Ragul29 I have escalated this to `andy@amazon.com` and received the following email:

> My name is Avery with the Executive Customer Relations team at AWS, following up on your message to Andy Jassy with the subject "Standalone S3 Lifecycle Rule for ExpiredObjectDeleteMarker=true does not work".
> 
> Thank you for notifying AWS of this issue. I wanted to let you know that I have contacted our operations team to review this request and take any necessary actions.
> 
> As no information is required from you, I will resolve this support case.
> 
> Best regards,
> Avery B.
> Amazon Web Services

**Comment by Ragul29** [comment]
Thanks for the update :) @urda
